tests/schema/tests.py

```diff
@@ -222,6 +224,49 @@ def test_fk(self):
             self.fail("No FK constraint for author_id found")
 
     @skipUnlessDBFeature('supports_foreign_keys')
+    def test_char_field_with_db_index_to_fk(self):
+        # Create the table
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+            editor.create_model(AuthorCharFieldWithIndex)
+        # Change CharField to FK
+        old_field = AuthorCharFieldWithIndex._meta.get_field("char_field")
+        new_field = ForeignKey(Author, CASCADE, blank=True)
+        new_field.set_attributes_from_name("char_field")
+        with connection.schema_editor() as editor:
+            editor.alter_field(AuthorCharFieldWithIndex, old_field, new_field, strict=True)
+        # Make sure the new FK constraint is present
+        constraints = self.get_constraints(AuthorCharFieldWithIndex._meta.db_table)
+        for name, details in constraints.items():
+            if details['columns'] == ["char_field_id"] and details['foreign_key']:
+                self.assertEqual(details['foreign_key'], ('schema_author', 'id'))
+                break
+        else:
+            self.fail("No FK constraint for author_id found")
+
+    @unittest.skipUnless(connection.vendor == 'postgresql', "PostgreSQL specific tests")
```

Comment:
I would like to use a database features combination
```python
@skipUnlessDBFeature('supports_foreign_keys')
@skipUnlessDBFeature('supports_index_on_textfield')
```
instead of 
```python
@unittest.skipUnless(connection.vendor == 'postgresql', "PostgreSQL specific tests") 
```
therefore I'm waiting for `supports_index_on_textfield` database feature (#8118).